### PR TITLE
Restore HTML syntax highlighting to PHP files

### LINKED ISSUE #11585: 8.4 Highlighting issues
### Description of the Issue
8.4 seems to bring issues with highlighting, PHP-files with HTML in them will not highlight the HTML as before. Also when downloading and editing PHP-files using the NppFTP-plugin the PHP will not highlight at all until switching tab or opening the Style configurator and clicking on PHP.

### Steps to Reproduce the Issue
Open a PHP file with HTML in it.

### Expected Behavior
Highlighted keywords.

### Actual Behavior
Highlighting missing.

### Debug Information
Notepad++ v8.4   (64-bit)
Build time : Apr 20 2022 - 03:31:06
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : C:\Users\Carl\OneDrive\Apps\Notepad++
OS Name : Windows 11 (64-bit) 
OS Version : 2009
OS Build : 22000.652
Current ANSI codepage : 1252
Plugins : NppFTP.dll

### LINKED ISSUE #11602: 8.4 PHP style is broken.
issue:
when viewing a .php file the colours for non-php-space change on an apostrophe. 

if teh text was within the php-space it would make sense to consider it the start of a PHP string, but it's occuring outside the <? ?> php-space.  it's occuring in the regular text space.

i don't recall having this issue in npp 8.3.x

### reproduce this issue
1.  ```<?php ?><html><body> isn't this awesome ..```
2.  observe how the `'` in the word  isn't  is being interpreted as a PHP string start token.
 

### Expected Behavior
it to know the difference between being inside and outside PHP <? ?> tags, when making style changes.

### Debug Information
Notepad++ v8.4   (64-bit)
Build time : Apr 20 2022 - 03:31:06
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Home (64-bit) 
OS Version : 2009
OS Build : 19044.1645
Current ANSI codepage : 1252
Plugins : JSMinNPP.dll mimeTools.dll NppConverter.dll NppExport.dll NPPJSONViewer.dll NppSnippets.dll NppXmlTreeviewPlugin.dll XMLTools.dll

### PR DESCRIPTION
**Pull request by rdipardo**
Fixes #11602 and (one part of) #11585 [^1]

[^1]: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11585#issuecomment-1111905407

### CODE REVIEW

File: PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp [hunk 1]
```diff
@@ -646,6 +646,7 @@ void ScintillaEditView::setXmlLexer(LangType type)
 	}
 	else if ((type == L_HTML) || (type == L_PHP) || (type == L_ASP) || (type == L_JSP))
 	{
+		setLexerFromLangID(L_HTML);
```
Comment (donho) [inline PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp:649]: Could you also make the following change:

```cpp
void ScintillaEditView::setXmlLexer(LangType type)
{
	if (type == L_XML)
	{
		setLexerFromLangID(L_XML);
		for (int i = 0 ; i < 4 ; ++i)
```

**Review by donho — CHANGES_REQUESTED** [review]
Please consider the suggested added modification.

File: PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp (hunk 1 above)
Comment (rdipardo) [reply PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp:649]: _Mais,regardez_ https://github.com/notepad-plus-plus/notepad-plus-plus/blob/3b0d5242acd23b4a727560ea98139cc5f4a8b88a/PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp#L634-L641

_Alors_, `type = L_XML`:

~~~cpp
	setLexerFromLangID(L_XML);

	if (L_XML == L_XML)
	{
		for (int i = 0 ; i < 4 ; ++i)
			execute(SCISETKEYWORDS, i, reinterpretcast<LPARAM>(TEXT("")));
	// ...
	}
~~~

npp-8-L_XML-detect

_Problème? où est-il?_

[^1]: https://github.com/notepad-plus-plus/notepad-plus-plus/blob/a61b03ea8887e21c6e1b7374068962f635b79b80/PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp#L631-L635

Comment (donho) [reply PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp:649]: What I want is make `setLexerFromLangID(L_XML);` inside of condition
```
if (type == L_XML)
{
```

so  `setLexerFromLangID` function won't be called twice in 
```cpp
else if ((type == L_HTML) || (type == L_PHP) || (type == L_ASP) || (type == L_JSP))
{
```

Comment (rdipardo) [reply PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp:649]: _Bien entendu_: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/11588/commits/cafafe1491a4e7a99489a246959e548da2b847e2

Comment (rdipardo) [reply PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp:649]: _Cette [dernière révision][0] est de Visual Studio. De nos jours, c'est en l'UTF-8 que se passe_:

~~~diff
@@ -1426,7 +1427,7 @@ void ScintillaEditView::defineDocType(LangType typeDoc)
        pStyle = stylers.findByID(defaultIndicatorStyle._styleID);
        setSpecialIndicator(pStyle ? *pStyle : defaultIndicatorStyle);
 
-    // Il faut surtout faire un test ici avant d'ex<E9>cuter SCI_SETCODEPAGE
+    // Il faut surtout faire un test ici avant d'exécuter SCI_SETCODEPAGE
     // Sinon y'aura un soucis de performance!
        if (isCJK())
        {
~~~
[0]: https://github.com/notepad-plus-plus/notepad-plus-plus/commit/cafafe1491a4e7a99489a246959e548da2b847e2#diff-d88ddee57a027ab23daf332c4778ced0cee352edcb34efdda1b218e8a75c61b2R1430

Comment (donho) [reply PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp:649]: Merci @rdipardo !

**Pull request closed without merging**